core/src/scene/stops.cpp

```diff
@@ -268,6 +323,32 @@ auto Stops::evalColor(float _key) const -> uint32_t {
     return Color::mix(lower->value.get<Color>(), upper->value.get<Color>(), lerp).abgr;
 }
 
+auto Stops::evalExpVec2(float _key) const -> glm::vec2 {
+    if (frames.empty()) { return glm::vec2{0.f}; }
+
+    auto upper = nearestHigherFrame(_key);
+    auto lower = upper - 1;
+
+    if (upper == frames.end()) {
+        return lower->value.get<glm::vec2>();
+    }
+    if (lower < frames.begin()) {
+        return upper->value.get<glm::vec2>();
+    }
+
+    double range = exp2(upper->key - lower->key) - 1.0;
+    double pos = exp2(_key - lower->key) - 1.0;
+
+    double lerp = pos / range;
+
+    const glm::vec2& lowerVal = lower->value.get<glm::vec2>();
+    const glm::vec2& upperVal = upper->value.get<glm::vec2>();
+
+    return glm::vec2(lowerVal.x * (1 - lerp) + upperVal.x * lerp,
+                     lowerVal.y * (1 - lerp) + upperVal.y * lerp);
+
+}
+
 auto Stops::evalVec2(float _key) const -> glm::vec2 {
```

Comment:
@karimnaaji  if the change should be linear with map scale this means it exponential with 'zoomlevel'